File: Review-06-Python-Shapes/Example-6/shapes/circle.py
import math
import copy

from shapes.shape import (Shape, FPT_FMT)

PI = math.pi  # \f$ \pi \f$
TAU = 2 * PI  # \f$ \tau = 2\pi \f$


class Circle(Shape):
    """
    An Ellipse with equivalent major and minor axes
    """

    def __init__(self, radius=1):
        """
        Construct a Circle
        """

        self._radius = radius

    @property
    def name(self) -> str:
        """
        Provide read-only access to the name attribute.
        """

        return "Circle"

    @property
    def radius(self):
        return self._radius

    @radius.setter
    def radius(self, some_value):
        self._radius = some_value

    @property
    def diameter(self):
        return 2 * self._radius

    def area(self):
        """
        Compute the area

        :return: area
        """
        return PI * (self._radius ** 2.0)

    def perimeter(self):
        """
        Compute the perimeter

        :return: perimeter
        """

        return TAU * self._radius

    def __deepcopy__(self, memo):
        """
        Return a new duplicate Shape
        """

        return Circle(copy.deepcopy(self.radius))

    def __str__(self):
        """
        Print the Circle
        """

        return (super().__str__()
                + FPT_FMT.format("Radius", self.radius)
                + FPT_FMT.format("Diameter", self.diameter)
                + FPT_FMT.format("Perimeter", self.perimeter())
                + FPT_FMT.format("Area", self.area()))

    def __repr__(self):
        return f"Circle(radius={self.radius})"

File: 06-Java-Shapes/Example-4/src/main/java/shapes/Circle.java
// Thomas Kennedy
// CS 330 Fall 2014

package shapes;

import java.util.Scanner;

/**
 * An Ellipse with equivalent major and minor axes
 */
public class Circle extends Shape
{
    private static final double PI  = Math.PI; ///< \f$ \pi \f$
    private static final double TAU = 2 * PI;  ///< \f$ \tau = 2\pi \f$

    private double _radius; ///< Length of the radius

    /**
     * Construct a Circle with radius set to 1
     */
    public Circle()
    {
        this._name   = "Circle";
        this._radius = 1;
    }

    /**
     * Construct a Circle
     *
     * @param s the desired radius length
     */
    public Circle(double r)
    {
        this._name   = "Circle";
        this._radius = r;
    }

    /**
     * Construct a Circle
     *
     * @param src the Circle to copy
     */
    public Circle(Circle src)
    {
        this._name   = src._name;
        this._radius = src._radius;
    }

    /**
     * Return the radius length
     */
    public double radius()
    {
        return _radius;
    }

    /**
     * Modify the radius length
     *
     * @param r the replacement length
     */
    public void radius(double r)
    {
        _radius = r;
    }

    /**
     * Return the diameter
     */
    public double diameter()
    {
        return 2 * _radius;
    }

    /**
     * Compute the area using \f$ \pi r^2 \f$
     *
     * @return area
     */
    @Override
    public double area()
    {
        return PI * _radius * _radius;
    }

    /**
     * Compute the perimeter using \f$ 2 \pi r \f$
     *
     * @return perimeter
     */
    @Override
    public double perimeter()
    {
        return TAU * _radius;
    }

    @Override
    public Shape clone()
    {
        return new Circle(this);
    }

    @Override
    public void read(Scanner scanner)
    {
        this._radius = scanner.nextDouble();
    }

    @Override
    public String toString()
    {
        return String.format(FMT_STR, "Name", this._name)
             + String.format(FMT_DBL, "Radius", this.radius())
             + String.format(FMT_DBL, "Diameter", this.diameter())
             + String.format(FMT_DBL, "Perimeter", this.perimeter())
             + String.format(FMT_DBL, "Area", this.area());
    }
}
